src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -424,6 +434,14 @@ internal void MarkThreadRequestSatisfied()
             // by the time we reach this point.
             //
             int count = numOutstandingThreadRequests;
+            if (dequeSuccessful && ThreadRequestNeeded(count))
```

Comment:
Also this condition is a bit different from #13408. I believe the idea was that if request count < proc count, then request count == work item count. So Dispatch would only need to request a thread if the request count == proc count, since at that point it doesn't know if there are more than proc count work items (and it only needs to request one thread in total). This condition would not request a thread if request count == proc count, that doesn't look right to me.